tst/ogb/main_pyg_with_pruning.py

```diff
@@ -135,35 +144,76 @@ def register_logging_files(args):
 
     return tb_writer, best_results_file, log_file
 
 
 def load_dataset(args):
     # automatic data loading and splitting
     transform = add_zeros if args.dataset == 'ogbg-ppa' else None
-    dataset = PygGraphPropPredDataset(name=args.dataset, transform=transform)
+    cls_criterion = get_loss_function(args.dataset)
+    idx2word_mapper = None
 
-    if args.dataset == 'obgb-code2':
-        seq_len_list = np.array([len(seq) for seq in dataset.data.y])
-        print('Target seqence less or equal to {} is {}%.'.format(args.max_seq_len,
-                                                                  np.sum(seq_len_list <= args.max_seq_len) / len(
-                                                                      seq_len_list)))
+    if args.dataset == 'mnist':
+        train_data = MNISTSuperpixels(root='dataset', train=True, transform=T.Polar())
+        dataset = train_data
+        dataset.name = 'mnist'
+        dataset.eval_metric = 'acc'
+        validation_data = []
+        test_data = MNISTSuperpixels(root='dataset', train=False, transform=T.Polar())
 
-    cls_criterion = get_loss_function(dataset.name)
-    idx2word_mapper = None
-    split_idx = dataset.get_idx_split()
-    # The following is only used in the evaluation of the ogbg-code classifier.
-    if args.dataset == 'ogbg-code2':
-        vocab2idx, idx2vocab = get_vocab_mapping([dataset.data.y[i] for i in split_idx['train']], args.num_vocab)
-        # specific transformations for the ogbg-code dataset
-        dataset.transform = transforms.Compose(
-            [augment_edge, lambda data: encode_y_to_arr(data, vocab2idx, args.max_seq_len)])
-        idx2word_mapper = partial(decode_arr_to_seq, idx2vocab=idx2vocab)
+        train_data = list(train_data)
+        test_data = list(test_data)
+
+    elif args.dataset == 'zinc':
+        train_data = ZINC(root='dataset', subset=True, split='train')
+
+        deg = torch.zeros(5, dtype=torch.long)
+        for data in train_data:
+            d = degree(data.edge_index[1], num_nodes=data.num_nodes, dtype=torch.long)
+            deg += torch.bincount(d, minlength=deg.numel())
+
+        dataset = train_data
+        dataset.name = 'zinc'
+        validation_data = ZINC(root='dataset', subset=True, split='val')
+        test_data = ZINC(root='dataset', subset=True, split='test')
+        dataset.eval_metric = 'mae'
+
+        train_data = list(train_data)
+        validation_data = list(validation_data)
+        test_data = list(test_data)
+
+    elif args.dataset in ['ogbg-molhiv', 'ogbg-molpcba', 'ogbg-ppa', 'ogbg-code2']:
+        dataset = PygGraphPropPredDataset(name=args.dataset, transform=transform)
 
-    return dataset, split_idx, cls_criterion, idx2word_mapper
+        if args.dataset == 'obgb-code2':
+            seq_len_list = np.array([len(seq) for seq in dataset.data.y])
+            print('Target seqence less or equal to {} is {}%.'.format(args.max_seq_len,
```

Comment:
nit: switch to using f-strings (`.format()` is Python 2.7 syntax). Also fix typo in "sequence".